### ISSUE TITLE
Game window + splash screen should be centered on game start

### ISSUE THREAD

**Issue by alexandervrs** [godotengine/godot#4430]
Tested on Windows. Mac & Linux possibly have this issue as well.

The game window (along with the splash screen window) on the Desktop platforms should be centered on game start for better UX+presentation. Currently I have to write a script to do that:

```gdscript
func _ready():

    var windowSize = Vector2(1280, 720);

    OS.set_window_resizable(false);
    OS.set_window_fullscreen(false);
    OS.set_window_size(windowSize);

    var displaySize = OS.get_screen_size();

    var windowX = (displaySize.width - windowSize.width)/2;
    var windowY = (displaySize.height - windowSize.height)/2;

    OS.set_window_position(Vector2(windowX, windowY));
```

_Using Godot 2.0.2_

**Comment by quantum-entangler** [comment]
And I assume you meant to add that it should happen on the main display? 'cause you know alternatives are either an alternative display or possibly in the middle of 2 screens depending on desktop configurations. ;)

**Comment by alexandervrs** [comment]
Yep, should be centered only on the main display!

**Comment by bojidar-bg** [comment]
I think this was fixed in master/2.1 through project settings, no?

**Comment by AlexHolly** [comment]
It was fixed in 2.1 and looks broken in 3.0

- Project selection spawns TopLeft
- Game window Spawns exactly centered and Ignores the Size of the game window.

![image](https://cloud.githubusercontent.com/assets/4741886/24961052/e5fc256c-1f97-11e7-83c7-a2efb44675e2.png)


Editorsettings->run->window placement->rect->centered



Tested on 3.0 Master Ubuntu

**Comment by akien-mga** [comment]
See #9418.
